### ISSUE TITLE
Missing `libgmp3-dev` for Ruby 2.2 on Debian 6

### ISSUE THREAD

**Issue by shoresh** [rvm/rvm#3675]
RVM install was fine.  When I tried to install Rails 2.2 it failed with:

E: Unable to locate package libgmp-dev

This was not surprising as libgmp-dev is not a valid package for Debian 6.  Unfortunately we are not in a position to upgrade to 7 at this time.  Is there a workaround or a version of rvm that I can use that is compatible with Debian 6?

Thanks

**Comment by cespo** [comment]
Same problem here:

`E: Unable to locate package libgmp-dev
++ /scripts/functions/utility : __rvm_try_sudo()  381 > return 100
++ /scripts/functions/requirements/debian : requirements_debian_libs_install()  36 > return 100
Requirements installation failed with status: 100.
make[1]: *** [install] Error 1`

**Comment by shoresh** [comment]
Unfortunately libgmp-dev is called libgmp3-dev in Debian 6 (or at least it
built itself that way when I tried to compile it from source).  I had to
get my sys admin to create a package that named it libgmp-dev and then
everything worked. You could also upgrade your Debian system to Debian 7 or
later, but I did not have that option.  There are some other options
involving patching the rvm package if you're interested -- sorry but I
can't recall the url.

**Comment by pkuczynski** [comment]
Same issue as #3680

**Comment by pkuczynski** [comment]
Is there still interest in getting this fixed? I might work on that this week...

**Comment by shoresh** [comment]
I can live with the monkey patch my sys admin provided me with, so I don't need a fix for Debian 6.  If the name mismatch is still happening in later versions of Debian than I would suggest that a fix is in order.  Thanks for following up on this.

**Comment by pkuczynski** [comment]
I would actually prefer to keep this open and apply a fix in case someone else will have similar issue. Just needed someone to help with testing, but I understand that you do not have time @shoresh?

**Comment by pkuczynski** [comment]
I prepared a patch. Could you test it and let me know if it works?

```
rvm get branch /bugfix/3824-libreadline
rvm install 2.1.4
```

**Comment by vStone** [comment]
works for me

**Comment by pkuczynski** [comment]
Perfect! I will get that merged as soon as possible.

**Comment by kenn** [comment]
I'm still seeing this error with `rvm 1.29.3 (latest)`. Is this fix deployed?

```
Error running 'requirements_debian_libs_install libsqlite3-dev sqlite3 libgmp-dev libgdbm-dev libtool bison pkg-config libffi-dev libgmp3-dev'
```

**Comment by kenn** [comment]
Oh, it looks like it's trying to install both `libgmp3-dev` AND `libgmp-dev`.

```
++ apt-get --no-install-recommends --yes install libsqlite3-dev sqlite3 libgmp-dev libgdbm-dev libtool bison pkg-config libffi-dev libgmp3-dev
```

**Comment by kenn** [comment]
I guess [this line](https://github.com/rvm/rvm/blob/1.29.3/scripts/functions/requirements/debian#L118) is causing the duplicate?

**Comment by kenn** [comment]
Just wanted to report that I hand-patched on the server and removed `libgmp-dev` in that line, and successfully installed ruby 2.3 / 2.4. Hope it helps.

**Comment by pkuczynski** [comment]
Nice catch @kenn. Fixed in #4238. `rvm get master` would help now.
